spring-kafka/src/main/java/org/springframework/kafka/core/CompositeKafkaConsumerProducerStrategy.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.core;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serializer;
+
+/**
+ * Composite implementation of Kafka Consumer and Producer strategy contract.
+ * <p>
+ * This {@link KafkaConsumerProducerStrategy} implementation will produce
+ * a new {@link KafkaConsumer} for provided {@link Map} {@code configs}, {@link Serializer} {@code keySerializer}
+ * and {@link Serializer} {@code valueSerializer} on {@link #createKafkaConsumer()} invocation.
+ * <p>
+ * This strategy will also produce {@link KafkaProducer} for provided {@link Map} {@code configs}, {@link Deserializer} {@code keyDeserializer}
+ * and {@link Deserializer} {@code valueDeserializer} on {@link #createKafkaProducer()} invocation.
+ *
+ * @param <K> the key type.
+ * @param <V> the value type.
+ *
+ * @author Murali Reddy
+ */
+public class CompositeKafkaConsumerProducerStrategy<K, V> implements KafkaConsumerProducerStrategy<K, V> {
```

Comment:
Well, @kvmkreddy ,

Thank you for the contribution one more time, first of all!

I am a bit nervous with this mix of concerns. 
My application can use only producer or consumer. But with this I have to configure something for both of them.
That really would be more clear just to have setters on the `DefaultKafkaConsumerFactory`/`DefaultKafkaProducerFactory`.

Even if you are going to have the common `configs` map, that is your architectural choice for your application.
But I'd prefer to distribute responsibility and don't worry about other independent components.

Your purpose of such a solution?